### PR TITLE
Add status id to films

### PR DESCRIPTION
**Pull request by brynyShift72**
[AB#5851](https://dev.azure.com/S72/fefacba9-96b6-4af2-a53d-050ed453e13a/_workitems/edit/5851) Adding Status ID to Film and TV Season models, will be used to differentiate published and non published content for admins.

### CODE REVIEW

**Comment by PeculiarGoat** [comment]
@brynyShift72 is this ready for review?

**Comment by brynyShift72** [comment]
> @brynyShift72 is this ready for review?

Yes should be ready, note that the original card says for pages and films, it turns out that pages + bundles will probably need some API changes and will be done later, this branch is just for Films + TV seasons :)

File: kibble/models/item_index.go [hunk 1]
```diff
@@ -83,6 +83,25 @@ func (itemIndex ItemIndex) Set(slug string, item GenericItem) {
 	}
 }
 
+// Set - an item
+func (itemIndex ItemIndex) SetWithStatus(slug string, statusID int, item GenericItem) {
```
Comment (PeculiarGoat) [inline kibble/models/item_index.go:87]: Just curious if there is a need to have both `Set` and `SetWithStatus`?

Comment (brynyShift72) [reply kibble/models/item_index.go:87]: There are a lot of uses of `itemIndex.Set` on items that do not really have an idea of Status ID yet, (although the generic item does) which I assume we would leave in until those items eventually have Status

**Review by PeculiarGoat — APPROVED** [review]
LGTM.  Better get a @stajs review too though.

File: kibble/models/generic_item.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ type GenericItem struct {
 	Title    string
 	Images   ImageSet
 	Seo      Seo
+	StatusID int
```
Comment (stajs) [inline kibble/models/generic_item.go:29]: What does this default to for items other than films and tv? nil?

What is the status/int map?

File: kibble/models/item_index.go [hunk 3]
```diff
@@ -83,6 +83,25 @@ func (itemIndex ItemIndex) Set(slug string, item GenericItem) {
...
+
+	slugType := getSlugType(slug)
+
+	index, ok := itemIndex[slugType]
+	if !ok {
+		itemIndex[slugType] = make(map[string]GenericItem)
+		index = itemIndex[slugType]
+	}
+
+	// unresolved can be overwritten, empty and item can not
```
Comment (stajs) [inline kibble/models/item_index.go:97]: The grammar on this comment threw me, I don't think we need the comment if we can potentially re-jig the `if`.

File: kibble/models/item_index.go [hunk 4]
```diff
@@ -83,6 +83,25 @@ func (itemIndex ItemIndex) Set(slug string, item GenericItem) {
...
+	foundItem, ok := index[slug]
+	item.StatusID = statusID
+	if (ok && (foundItem == Unresolved || foundItem == Empty)) || !ok {
```
Comment (stajs) [inline kibble/models/item_index.go:100]: Is this the same?

```suggestion
	if !ok || foundItem == Unresolved || foundItem == Empty {
```

File: kibble/models/item_index.go [hunk 5]
```diff
@@ -125,14 +144,19 @@ func (itemIndex ItemIndex) findSlugsOfType(slugType string, itemType GenericItem
...
+			if deepCompare(v, itemType) {
 				found = append(found, k)
 			}
 		}
 	}
 	return found
 }
 
+//Check fields of GenericItem, ignore StatusID (publish status) as it may differ.
+func deepCompare(a GenericItem, b GenericItem) bool {
```
Comment (stajs) [inline kibble/models/item_index.go:156]: `deepCompare()` sounds like it's recursing properties or something. How about `equalsIgnoreStatus()`?

**Review by stajs — CHANGES_REQUESTED** [review]
Minor things, mostly questions.

File: kibble/models/item_index.go (hunk 4 above)
Comment (brynyShift72) [reply kibble/models/item_index.go:100]: Also from the Set function, but yeah that way does seem a lot more readable, will update

File: kibble/models/item_index.go (hunk 3 above)
Comment (brynyShift72) [reply kibble/models/item_index.go:97]: Whoops that comment is copied over from the original version of this function in `Set()`

File: kibble/models/generic_item.go (hunk 2 above)
Comment (brynyShift72) [reply kibble/models/generic_item.go:29]: It seems its defaults to 0.
kibble doesn't really care about these values too much in terms of logic, but 1 = unpublished and 2 = published. It'll be 0 for all other generic items

**Pull request merged**